File: leetcode/203.移除链表元素.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeElements(head *ListNode, val int) *ListNode {
    root := &ListNode{0, head}
    cur := head
    pre := root
    for cur != nil {
        if cur.Val == val {
            pre.Next = cur.Next
            cur = cur.Next
        } else {
            pre = pre.Next
            cur = cur.Next
        }
    }
    return root.Next
}

File: leetcode/203.移除链表元素.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        root=ListNode(0,head)
        cur=root.next
        pre=root
        while cur:
            if cur.val==val:
                cur=cur.next
                del pre.next
                pre.next=cur
            else:
                cur=cur.next
                pre=pre.next
        return root.next
                